backends/s3/s3.go

```diff
@@ -169,57 +169,64 @@ func (b *Backend) doList(ctx context.Context, prefix string) (simpleblob.BlobLis
 	return blobs, nil
 }
 
+// Load retrieves the content of the object identified by name from S3 Bucket
+// configured in b.
 func (b *Backend) Load(ctx context.Context, name string) ([]byte, error) {
 	metricCalls.WithLabelValues("load").Inc()
 	metricLastCallTimestamp.WithLabelValues("load").SetToCurrentTime()
 
 	obj, err := b.client.GetObject(ctx, b.opt.Bucket, name, minio.GetObjectOptions{})
-	if err != nil {
-		if err = handleErrorResponse(err); err != nil {
-			return nil, err
-		}
+	if err = convertMinioError(err); err != nil {
+		return nil, err
 	} else if obj == nil {
 		return nil, os.ErrNotExist
 	}
 
 	p, err := io.ReadAll(obj)
-	if err = handleErrorResponse(err); err != nil {
+	if err = convertMinioError(err); err != nil {
 		return nil, err
 	}
 	return p, nil
 }
 
+// Store sets the content of the object identified by name to the content
+// of data, in the S3 Bucket configured in b.
 func (b *Backend) Store(ctx context.Context, name string, data []byte) error {
-	if err := b.doStore(ctx, name, data); err != nil {
+	info, err := b.doStore(ctx, name, data)
+	if err != nil {
 		return err
 	}
-	if b.opt.UseUpdateMarker {
-		if err := b.doStore(ctx, UpdateMarkerFilename, []byte(name)); err != nil {
-			return err
-		}
-	}
-	return nil
+	return b.setMarker(ctx, name, info.ETag, false)
 }
 
-func (b *Backend) doStore(ctx context.Context, name string, data []byte) error {
+func (b *Backend) doStore(ctx context.Context, name string, data []byte) (*minio.UploadInfo, error) {
 	metricCalls.WithLabelValues("store").Inc()
 	metricLastCallTimestamp.WithLabelValues("store").SetToCurrentTime()
 
-	_, err := b.client.PutObject(ctx, b.opt.Bucket, name, bytes.NewReader(data), int64(len(data)), minio.PutObjectOptions{
+	info, err := b.client.PutObject(ctx, b.opt.Bucket, name, bytes.NewReader(data), int64(len(data)), minio.PutObjectOptions{
 		NumThreads: 3,
 	})
 	if err != nil {
 		metricCallErrors.WithLabelValues("store").Inc()
 	}
-	return err
+	return &info, err
```

Comment:
If you need to take the address of this to convert it into the `*minio.UploadInfo`, you probably should just return `minio.UploadInfo` instead.